tests/components/squeezebox/conftest.py

```diff
@@ -115,20 +131,157 @@ async def mock_async_browse(
-    player.player_id = TEST_MAC
-    player.name = "Test Player"
-    player.power = False
+
+    # set some default fake values for properties
+    player.player_id = random_uuid_hex()
+    player.current_index = 0
+    player.mode = "stop"
+    player.muting = False
+    player.name = TEST_PLAYER_NAME
+    player.playlist: list(dict(str, str)) | None = None
+    player.power = True
+    player.repeat = "none"
+    player.shuffle = "none"
+    player.time = 0
+    player.volume = 10
+
+    # mock pysqueezebox.Player methods
     player.async_browse = AsyncMock(side_effect=mock_async_browse)
-    player.async_load_playlist = AsyncMock()
+    player.async_clear_playlist = AsyncMock(
+        side_effect=lambda: setattr(player, "playlist", None)
+    )
+    player.async_index = AsyncMock(
+        side_effect=lambda index: setattr(
+            player,
+            "current_index",
+            player.current_index + int(index)
+            if index.startswith(("+", "-"))
+            else index,
+        )
+    )
+
+    def async_load_playlist(playlist, cmd):
+        if cmd == "add":
+            if player.playlist is None:
+                player.playlist = []
+            player.playlist.extend(playlist)
+        else:
+            player.playlist = playlist
+        player.current_index = 0
+        player.url = player.playlist[0]
+
+    player.async_load_playlist = AsyncMock(side_effect=async_load_playlist)
+    player.async_load_url = AsyncMock(
+        side_effect=lambda url, cmd: async_load_playlist([{"url": url}], cmd)
+    )
+    player.async_play = AsyncMock(side_effect=lambda: setattr(player, "mode", "play"))
+    player.async_pause = AsyncMock(side_effect=lambda: setattr(player, "mode", "pause"))
+    player.async_set_muting = AsyncMock(
+        side_effect=lambda muting: setattr(player, "muting", muting)
+    )
+    player.async_set_power = AsyncMock(
+        side_effect=lambda power: setattr(player, "power", power)
+    )
+    player.async_set_repeat = AsyncMock(
+        side_effect=lambda repeat: setattr(player, "repeat", repeat)
+    )
+    player.async_set_shuffle = AsyncMock(
+        side_effect=lambda shuffle: setattr(player, "shuffle", shuffle)
+    )
+    player.async_set_volume = AsyncMock(
+        side_effect=lambda volume: setattr(
+            player,
+            "volume",
+            player.volume + int(volume) if volume.startswith(("+", "-")) else volume,
+        )
+    )
+    player.async_stop = AsyncMock(side_effect=lambda: setattr(player, "mode", "stop"))
+    player.async_sync = AsyncMock()
+    player.async_time = AsyncMock(
+        side_effect=lambda time: setattr(player, "time", time)
+    )
+    player.async_toggle_pause = AsyncMock(
+        side_effect=lambda: setattr(
+            player, "mode", "play" if player.mode != "play" else "pause"
+        )
+    )
+    player.async_query = AsyncMock(return_value="test result")
     player.async_update = AsyncMock()
+    player.async_unsync = AsyncMock()
```

Comment:
I personally think we can improve the mocking and tests by not having the mocks try to imitate the library. Because (afaict) the integration does not depend on the integration taking state after a call, so I think it's more valuable to only check if the mock is called correctly. This way your tests become a bit more loose. I think we also can make use of autospec. Check AirGradient for an example. With autospec you don't need to completely mock all fields of a mock, but every valid method or property becomes an AsyncMock automatically. Wdyt?

Reply:
Sure, I'm good with that approach. I'll mark this as draft while I make the changes.

Reply:
Ok, this is the approach used now (autospec the integration). I checked the calls, and separately tested retrieving property values to cover that part of the code.